src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp

```diff
@@ -323,7 +323,26 @@ void SetExeNetworkInfo(const std::shared_ptr<IExecutableNetworkInternal>& exeNet
     const auto& inputsInfo = exeNetwork->GetInputsInfo();
     const auto& outputsInfo = exeNetwork->GetOutputsInfo();
     OPENVINO_ASSERT(inputsInfo.size() == function->get_parameters().size());
-    OPENVINO_ASSERT(outputsInfo.size() == function->get_output_size());
+
+    if (outputsInfo.size() != function->get_output_size()) {
+        std::unordered_set<std::string> output_names;
```

Comment:
Thank you for the suggestion, I've refactored the code to use such approach, but with tensor pointers instead of names.
Please take a look at the latest version.